### PR TITLE
Added new contracts for credentialsfetcher

### PR DESCRIPTION
**Pull request by saikiranakula-amzn**
1. Add support to retrieve from s3/ssm
2. Added flags to have the support enabled only in Amazon Linux

*Issue #, if available:*

*Description of changes:*
[ec2-user@EC2AMAZ-7ZXeXp build]$ sudo ./credentials-fetcherd 
krb_files_dir = /var/credentials-fetcher/krbdir
cred_file =  (lease id: )
logging_dir = /var/credentials-fetcher/logging
unix_socket_dir = /var/credentials-fetcher/socket
Thread 0: top of stack near 0x7f338a878a18; argv_string=grpc_thread
Thread 0: top of stack near 0x7f3382077968; argv_string=krb_ticket_refresh_thread
Server listening on unix:/var/credentials-fetcher/socket/credentials_fetcher.sock
CallDataCreateKerberosLease 0x7f3384009a90 status: 0
AddNonDomainJoinedKerberosLease 0x7f338400a2a0 status: 0
RenewNonDomainJoinedKerberosLease 0x7f3384006ad0 status: 0
CallDataDeleteKerberosLease 0x7f3384007470 status: 0
CallHealthCheck 0x7f3384007e30 status: 0
CallDataCreateKerberosArnLease 0x7f3384024f40 status: 0
RenewKerberosArnLease 0x7f3384025910 status: 0

[ec2-user@EC2AMAZ-7ZXeXp tests]$ sudo ./gmsa_test_client --create_kerberos_tickets_non_domain_joined xxxxx xxxxx contoso.com
krb tickets will get created
created ticket file /var/credentials-fetcher/krbdir/b7bbc7ff9379cbe6a624/WebApp01
created ticket file /var/credentials-fetcher/krbdir/b7bbc7ff9379cbe6a624/WebApp03
Client received output for add kerberos lease non domain joined: b7bbc7ff9379cbe6a624

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: api/src/gmsa_service.cpp [hunk 1]
```diff
@@ -227,6 +242,548 @@ class CredentialsFetcherImpl final
...
+                    for (int i = 0;
+                         i < create_arn_krb_request_.credspec_arns_size(); i++) {
+                        creds_fetcher::krb_ticket_info *krb_ticket_info =
+                                new creds_fetcher::krb_ticket_info;
+                        creds_fetcher::krb_ticket_arn_mapping *krb_ticket_arns =
+                                new creds_fetcher::krb_ticket_arn_mapping;
+
+                        // get credentialspec contents:
+                        Aws::Auth::AWSCredentials creds = get_credentials(accessId, secretKey, sessionToken);
+                        std::string response = retrieve_credspec_from_s3(create_arn_krb_request_.credspec_arns(i), region, creds, false);
```
Comment (smhmhmd) [inline api/src/gmsa_service.cpp:323]: Add a check for creds and response

File: CMakeLists.txt [hunk 2]
```diff
@@ -159,6 +164,9 @@ install(FILES ${CMAKE_BINARY_DIR}/credentials_fetcher_utf16_private.exe
 install(FILES ${CMAKE_BINARY_DIR}/credentials_fetcher_utf16_private.runtimeconfig.json
         DESTINATION "/usr/sbin/"
         PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ)
+install(FILES ${CMAKE_BINARY_DIR}/krb5.conf
+        DESTINATION "/usr/sbin/"
```
Comment (smhmhmd) [inline CMakeLists.txt:168]: krb5.conf needs to be /etc

File: api/src/gmsa_service.cpp (hunk 1 above)
Comment (saikiranakula-amzn) [reply api/src/gmsa_service.cpp:323]: added check for response

File: CMakeLists.txt (hunk 2 above)
Comment (saikiranakula-amzn) [reply CMakeLists.txt:168]: will take a look at it in subsequent PR

**Review by smhmhmd — APPROVED** [review]
Keep track of the krb5 change for later

**Pull request merged**